src/__template__/ComponentTemplate.vue

```diff
@@ -90,6 +99,48 @@ const rootElementStyles = computed((): Record<string, string> => ({
   lineHeight: '0',
   width: iconSize.value,
 }))
+
+/**
+ * Prefix all SVG IDs in the given SVG string with a random prefix to ensure uniqueness.
+ *
+ * This function performs the following steps:
+ * 1. Generates a random prefix.
+ * 2. Replaces all `id` attributes in the SVG string with the new prefixed IDs.
+ * 3. Updates all ID references (e.g., `url(#id)`, `href="#id"`, etc.) to use the new prefixed IDs.
+ *
+ * @param {string} svgString - The SVG string in which to prefix IDs.
+ * @returns {string} - The SVG string with prefixed IDs.
+ */
+const prefixSvgIdsInString = (svgString: string): string => {
+  const idMap: Record<string, string> = {}
+  const randomPrefix = Math.random().toString(36).substring(2, 12)
+
+  // Replace IDs in the SVG string
+  const updatedSvgString = svgString.replace(/id="([^"]+)"/g, (match, originalId) => {
+    const newId = `${randomPrefix}-${originalId}`
+    idMap[originalId] = newId
+    return `id="${newId}"`
+  })
+
+  // Replace ID references (e.g., url(#id), href="#id", etc.)
+  const processedSvgString = updatedSvgString
+    .replace(/url\(#([^)]+)\)/g, (match, originalId) => idMap[originalId] ? `url(#${idMap[originalId]})` : match)
+    .replace(/#([^\s]+)/g, (match, originalId) => idMap[originalId] ? `#${idMap[originalId]}` : match)
+
+  // Return the processed SVG string
+  return processedSvgString
+}
+
+// Initially set `svgOriginalContent` to a template string. We will then do a dynamic replacement of any ids in the SVG string.
+// eslint-disable-next-line @stylistic/quotes
+const svgOriginalContent = ref<string>(`{%%ICON_SVG_INNER_HTML%%}`)
+const svgProcessedContent = computed((): string => {
```

Comment:
nitpick: not sure whether we want this to be reactive, I cannot think of any use case that we may want to change the behavior dynamically.